### PR TITLE
mvcc: add lock ttl.

### PR DESCRIPTION
**Pull request by disksing**
Make it possible for the driver to set lock ttl for large transactions.

@siddontang @zhangjinpeng1987

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

File: src/storage/mvcc/lock.rs [hunk 1]
```diff
@@ -85,6 +88,11 @@ impl Lock {
         let lock_type = try!(LockType::from_u8(try!(b.read_u8())).ok_or(Error::BadFormatLock));
         let primary = try!(b.decode_compact_bytes());
         let ts = try!(b.decode_var_u64());
-        Ok(Lock::new(lock_type, primary, ts))
+        let ttl = if b.len() == 0 {
+            0
```
Comment (siddontang) [inline src/storage/mvcc/lock.rs]: why 0 here?

**Comment by siddontang** [comment]
LGTM

**Pull request merged**